File: Examples/VisualBasic/ModifyingAndConvertingImages/SplittingTiffFrames.vb
Imports System.IO
Imports Aspose.Imaging.FileFormats.Tiff
Imports Aspose.Imaging.FileFormats.Tiff.Enums
Imports Aspose.Imaging.ImageOptions
Imports Aspose.Imaging.Sources

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'


Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages
    Class SplittingTiffFrames
        Public Shared Sub Run()
            ' ExStart:SplittingTiffFrames
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_ModifyingAndConvertingImages()

            ' Create an instance of TiffImage and load the file from disc
            Using multiImage = DirectCast(TiffImage.Load(dataDir & Convert.ToString("SampleTiff1.tiff")), TiffImage)
                'Initialize a variable to keep track of the frames in the image
                Dim i As Integer = 0
                ' Iterate over the tiff frame collection
                For Each tiffFrame In multiImage.Frames
                    ' Save the frame directly on disc in Jpeg compression
                    tiffFrame.Save((dataDir & System.Math.Max(System.Threading.Interlocked.Increment(i), i - 1)) + ".tiff", New TiffOptions(TiffExpectedFormat.TiffJpegRgb))
                Next
            End Using
            ' ExEnd:SplittingTiffFrames
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/SplittingTiffFrames.cs
// GIST-ID: f6899572dec565b5543b1e3a28eeade8
using Aspose.Imaging.FileFormats.Tiff;
using Aspose.Imaging.FileFormats.Tiff.Enums;
using Aspose.Imaging.ImageOptions;
using System;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API
when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information.
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET from https://releases.aspose.com/,
install it, and then add its reference to this project. For any issues, questions, or suggestions,
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    class SplittingTiffFrames
    {
        public static void Run()
        {
            Console.WriteLine("Running example SplittingTiffFrames");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();

            // Create an instance of TiffImage and load the file from disk.
            using (var multiImage = (TiffImage)Image.Load(dataDir + "SampleTiff1.tiff"))
            {
                // Initialize a variable to keep track of the frames in the image.
                // Iterate over the TIFF frame collection and save each image.
                int i = 0;
                foreach (var tiffFrame in multiImage.Frames)
                {
                    tiffFrame.Save(dataDir + i + "_out.tiff", new TiffOptions(TiffExpectedFormat.TiffJpegRgb));
                }
            }

            Console.WriteLine("Finished example SplittingTiffFrames");
        }
    }
}